test.js

```diff
@@ -106,42 +109,275 @@ test('allow not removing empty dependency properties sync', t => {
+	t.deepEqual(Object.keys(packageJson.peerDependencies), ['bar', 'foo']);
+});
+
+test('addPackageDependencies - sync - two types with empty', t => {
+	const temporary = temporaryDirectory();
+	writePackageSync(temporary, pick(fixture, ['foo', 'scripts', 'dependencies', 'devDependencies']));
+	addPackageDependenciesSync(temporary, pick(addFixture, ['dependencies', 'devDependencies']));
+	const packageJson = readPackageSync({cwd: temporary, normalize: false});
+
+	t.true(packageJson.foo);
+	t.deepEqual(Object.keys(packageJson.scripts), ['b', 'a']);
+	t.deepEqual(Object.keys(packageJson.dependencies), ['bar', 'baz', 'foo']);
+	t.deepEqual(Object.keys(packageJson.devDependencies), ['bar', 'baz', 'foo']);
+	t.falsy(packageJson.optionalDependencies);
+	t.falsy(packageJson.peerDependencies);
+});
+
+test('addPackageDependencies - sync - overwrite dependency', t => {
+	const temporary = temporaryDirectory();
+	writePackageSync(temporary, fixture);
+	addPackageDependenciesSync(temporary, {bar: '2.0.0'});
+	const packageJson = readPackageSync({cwd: temporary, normalize: false});
+
+	t.deepEqual(Object.keys(packageJson.dependencies), ['bar', 'foo']);
+	t.is(packageJson.dependencies.bar, '2.0.0');
+});
+
+test('addPackageDependencies - sync - allow not removing empty dependency properties', t => {
+	const temporary = temporaryDirectory();
+	writePackageSync(temporary, emptyPropFixture, {normalize: false});
+	addPackageDependenciesSync(temporary, pick(addFixture, ['dependencies', 'devDependencies']), {normalize: false});
+	const packageJson = readPackageSync({cwd: temporary, normalize: false});
+
+	t.true(packageJson.foo);
+	t.deepEqual(Object.keys(packageJson.dependencies), ['baz']);
+	t.deepEqual(Object.keys(packageJson.devDependencies), ['baz']);
+	t.truthy(packageJson.optionalDependencies);
+	t.truthy(packageJson.peerDependencies);
+});
+
+test('addPackageDependencies - sync - create package.json if one does not exist', t => {
+	const temporary = temporaryDirectory();
+	addPackageDependenciesSync(temporary, addFixture);
+	const {
+		dependencies,
+		devDependencies,
+		optionalDependencies,
+		peerDependencies,
+		...rest
+	} = readPackageSync({cwd: temporary, normalize: false});
+
+	t.deepEqual(Object.keys(dependencies), ['baz']);
+	t.deepEqual(Object.keys(devDependencies), ['baz']);
+	t.deepEqual(Object.keys(optionalDependencies), ['baz']);
+	t.deepEqual(Object.keys(peerDependencies), ['baz']);
+	t.true(Object.keys(rest).length === 0, 'package.json had additional fields!');
+});
+
+const missingEndingBraceFixture = '{"name": "foo", "dependencies": {"bar": "1.0.0"}';
+
+test('addPackageDependencies - async - invalid package.json', async t => {
+	const temporary = temporaryDirectory();
+	await fsPromises.writeFile(path.join(temporary, 'package.json'), missingEndingBraceFixture);
+
+	await t.throwsAsync(
+		addPackageDependencies(temporary, addFixture.dependencies),
+		{name: 'JSONError'},
+	);
```

Comment:
Should this (and `writePackage`) have a custom error message for this?

Reply:
I don't think that's needed.